Osx build fails with multiple undefined symbols errors while linking executables using clang compiler

Since merging #6227, a bug that's been hidden under the hood, appeared on the surface. The build fails on osx with clang compiler, failing when linking executables. The commit that first causes it is 85511a642f851100d0d856f4ecbe7fea7a7bb049
Branch for osx build buxfix is updated with latest upstream version, including travis fix, with my other commits on top. The error flow as shown on on travis:
```
[ 84%] Linking CXX executable ../../bin/minetestserver
Undefined symbols for architecture x86_64:
  "NetworkPacket::operator<<(unsigned long)", referenced from:
      Server::handleCommand_Init_Legacy(NetworkPacket*) in serverpackethandler.cpp.o
      Server::SendCSMFlavourLimits(unsigned short) in server.cpp.o
      Server::acceptAuth(unsigned short, bool) in server.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [../bin/minetestserver] Error 1
make[1]: *** [src/CMakeFiles/minetestserver.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
...
[100%] Linking CXX executable ../../bin/minetest
ld: warning: object file (/usr/local/lib/libIrrlicht.a(Irrlicht.o)) was built for newer OSX version (10.12) than being linked (10.11)
... (multiple warnings repeating) ...
ld: warning: object file (/usr/local/lib/libIrrlicht.a(jidctfst.o)) was built for newer OSX version (10.12) than being linked (10.11)
Undefined symbols for architecture x86_64:
  "NetworkPacket::operator<<(unsigned long)", referenced from:
      Server::handleCommand_Init_Legacy(NetworkPacket*) in serverpackethandler.cpp.o
      Server::SendCSMFlavourLimits(unsigned short) in server.cpp.o
      Server::acceptAuth(unsigned short, bool) in server.cpp.o
  "NetworkPacket::operator>>(unsigned long&)", referenced from:
      Client::handleCommand_AuthAccept(NetworkPacket*) in clientpackethandler.cpp.o
      Client::handleCommand_InitLegacy(NetworkPacket*) in clientpackethandler.cpp.o
      Client::handleCommand_CSMFlavourLimits(NetworkPacket*) in clientpackethandler.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [../bin/minetest] Error 1
make[1]: *** [src/CMakeFiles/minetest.dir/all] Error 2
make: *** [all] Error 2
```

--- a/src/network/networkpacket.h
+++ b/src/network/networkpacket.h
@@ -19,10 +19,10 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #pragma once
 
-#include <SColor.h>
 #include <ctime>
 #include "util/pointer.h"
 #include "util/numeric.h"
+#include <SColor.h>
 
 class NetworkPacket
 {

